_pytest/_code/code.py

```diff
@@ -354,7 +354,10 @@ def __init__(self, tup=None, exprinfo=None):
             if exprinfo is None and isinstance(tup[1], AssertionError):
                 exprinfo = getattr(tup[1], 'msg', None)
                 if exprinfo is None:
-                    exprinfo = str(tup[1])
+                    try:
+                        exprinfo = str(tup[1])
+                    except UnicodeEncodeError:
+                        exprinfo = unicode(tup[1])
```

Comment:
Actually, `tup[1]` is an `AssertionError` instance containing a `unicode` string. I think the proper fix is actually using `py._builtin._totext`:

```diff
diff --git a/_pytest/_code/code.py b/_pytest/_code/code.py
index 040e4fc..9a6d326 100644
--- a/_pytest/_code/code.py
+++ b/_pytest/_code/code.py
@@ -354,7 +354,7 @@ class ExceptionInfo(object):
             if exprinfo is None and isinstance(tup[1], AssertionError):
                 exprinfo = getattr(tup[1], 'msg', None)
                 if exprinfo is None:
-                    exprinfo = str(tup[1])
+                    exprinfo = py._builtin._totext(tup[1])
                 if exprinfo and exprinfo.startswith('assert '):
                     self._striptext = 'AssertionError: '
         self._excinfo = tup
```

`py._builtin._totext` is just `unicode` in Python 2 and `str` in Python 3. With the above change the test passes to me.